src/mon/OSDMonitor.cc

```diff
@@ -4764,9 +4764,17 @@ void OSDMonitor::check_osdmap_subs()
 void OSDMonitor::check_osdmap_sub(Subscription *sub)
 {
   dout(10) << __func__ << " " << sub << " next " << sub->next
-	   << (sub->onetime ? " (onetime)":" (ongoing)") << dendl;
+	   << (sub->onetime ? " (onetime)":" (ongoing)")
+	   << (sub->want_latest ? " (noosd)":" (osd)")
+	   << " epoch: " << osdmap.get_epoch()
+	   << dendl;
+  
   if (sub->next <= osdmap.get_epoch()) {
-    if (sub->next >= 1)
+    if (sub->next >= 1 &&
+	(!sub->want_latest ||
+	 g_conf().get_val<uint64_t>("mon_max_span_of_incremental") == 0 ||
```

Comment:
g_conf().get_val() is an expensive function. should avoid calling it multiple times

Reply:
updated it